src/components/action-bar/action-bar.types.ts

```diff
@@ -1,9 +1,16 @@
 import { MenuItem } from '../menu/menu.types';
 
-export interface ActionBarItem extends MenuItem {
-    /**
-     * Renders the button in the action bar without their labels.
-     * Does not affect the items that are overflown into the overflow menu.
-     */
-    iconOnly?: boolean;
+/**
+ * Renders the button in the action bar without their labels.
+ * Does not affect the items that are overflown into the overflow menu.
+ */
+export type ActionBarItem = ActionBarItemWithLabel | ActionBarItemWithoutLabel;
+
+interface ActionBarItemWithLabel extends MenuItem {
```

Comment:
Now you swapped them again 🤣 
```suggestion
interface ActionBarItemOnlyIcon extends MenuItem {
```